echopype/commongrid/api.py

```diff
@@ -0,0 +1,217 @@
+"""
+Functions for enhancing the spatial and temporal coherence of data.
+"""
+
+import numpy as np
+import pandas as pd
+import xarray as xr
+
+from ..utils.prov import echopype_prov_attrs
+from .mvbs import get_MVBS_along_channels
+
+
+def _set_MVBS_attrs(ds):
+    """
+    Attach common attributes.
+
+    Parameters
+    ----------
+    ds : xr.Dataset
+        dataset containing MVBS
+    """
+    ds["ping_time"].attrs = {
+        "long_name": "Ping time",
+        "standard_name": "time",
+        "axis": "T",
+    }
+
+    ds["Sv"].attrs = {
+        "long_name": "Mean volume backscattering strength (MVBS, mean Sv re 1 m-1)",
+        "units": "dB",
+        "actual_range": [
+            round(float(ds["Sv"].min().values), 2),
+            round(float(ds["Sv"].max().values), 2),
+        ],
+    }
+
+
+def compute_MVBS(ds_Sv, range_meter_bin=20, ping_time_bin="20S"):
+    """
+    Compute Mean Volume Backscattering Strength (MVBS)
+    based on intervals of range (``echo_range``) and ``ping_time`` specified in physical units.
+
+    Output of this function differs from that of ``compute_MVBS_index_binning``, which computes
+    bin-averaged Sv according to intervals of ``echo_range`` and ``ping_time`` specified as
+    index number.
+
+    Parameters
+    ----------
+    ds_Sv : xr.Dataset
+        dataset containing Sv and ``echo_range`` [m]
+    range_meter_bin : Union[int, float]
+        bin size along ``echo_range`` in meters, default to ``20``
+    ping_time_bin : str
+        bin size along ``ping_time``, default to ``20S``
+
+    Returns
+    -------
+    A dataset containing bin-averaged Sv
+    """
+
+    # create bin information for echo_range
+    range_interval = np.arange(0, ds_Sv["echo_range"].max() + range_meter_bin, range_meter_bin)
+
+    # create bin information needed for ping_time
+    ping_interval = (
+        ds_Sv.ping_time.resample(ping_time=ping_time_bin, skipna=True).asfreq().ping_time.values
+    )
+
+    # calculate the MVBS along each channel
+    MVBS_values = get_MVBS_along_channels(ds_Sv, range_interval, ping_interval)
+
+    # create MVBS dataset
+    ds_MVBS = xr.Dataset(
+        data_vars={"Sv": (["channel", "ping_time", "echo_range"], MVBS_values)},
+        coords={
+            "ping_time": ping_interval,
+            "channel": ds_Sv.channel,
+            "echo_range": range_interval[:-1],
+        },
+    )
+
+    # Added this check to support the test in test_process.py::test_compute_MVBS
+    if "filenames" in ds_MVBS.variables:
+        ds_MVBS = ds_MVBS.drop_vars("filenames")
```

Comment:
Do you remember what this is? It's ok if not. I think the MVBS code needs to be looked at further as well anyways.

Reply:
I don't remember.

Reply:
```suggestion
    # TODO: look into why 'filenames' exist here as a variable
    # Added this check to support the test in test_process.py::test_compute_MVBS
    if "filenames" in ds_MVBS.variables:
        ds_MVBS = ds_MVBS.drop_vars("filenames")
```
Let's add a TODO here.